### DISCUSSION TITLE
How to model linear travel with different friction regimes?

### DISCUSSION THREAD

**Discussion by faisal-shah** [Hopsan/hopsan#2210] [Q&A]
A piston is driving a mass with friction. However, for the first half of the travel there is a particular friction profile, and for the second half of the travel there is a different friction profile. If there was a way to place a 'gap' between two mass with friction components I can see how to transition from a lower friction to a higher (they would add up when collide and start moving together). How would this be accomplished?
But, how about going from a higher friction to a lower one?

**Comment by robbr48** [comment]
This should be possible, the stiffness parameter in the spring component is of "input interface" type, meaning that you can convert it to an input port. This makes it possible to set the spring stiffness to zero whenever there shuould be no contact between the two masses, based on the signal from a position sensor. 

I created a little example model. The lower bound limitation for mass 2 is also important.

[dualfriction.zip](https://github.com/Hopsan/hopsan/files/15149539/dualfriction.zip)
